lib/jsdom/living/helpers/geometry/dommatrix-init.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+
+const { matrix2dArrayProperties, matrix3dArrayProperties } = require("./matrix");
+
+const aliasedProperties = [
+  ["a", "m11", 1],
+  ["b", "m12", 0],
+  ["c", "m21", 0],
+  ["d", "m22", 1],
+  ["e", "m41", 0],
+  ["f", "m42", 0]
+];
+
+// Properties that are required to be zero for a valid 2D Matrix
+const matrix2dZeroProps = [
+  "m13", "m14", "m23", "m24",
+  "m31", "m32", "m34", "m43"
+];
+
+// Properties that are required to be one for a valid 2D matrix
+const matrix2dOneProps = ["m33", "m44"];
+
+
+function hasProp(o, prop) {
+  return Object.prototype.hasOwnProperty.call(o, prop);
+}
+
+function propertyIsTrue(o, prop) {
+  return hasProp(o, prop) && Boolean(o[prop]);
+}
+
+// If both alias and actual properties are present, they must be equal.
+function validateAliasedProps(init) {
+  aliasedProperties.forEach(([alias, prop]) => {
+    if (
+      hasProp(init, alias) &&
+      hasProp(init, prop) &&
+      init[alias] !== init[prop]
+    ) {
+      throw new TypeError(`Invalid Matrix initialization object.`);
+    }
+  });
+}
+
+function isValid2dMatrix(init) {
+  return !(matrix2dZeroProps.some(x => hasProp(init, x) && init[x] !== 0) ||
+    matrix2dOneProps.some(x => hasProp(init, x) && init[x] !== 1));
+}
+
+function getDOMMatrixDimension(init) {
+  if (hasProp(init, "id2D") && !init.is2D) {
```

Comment:
For any internal function, we can assume that the input is "well-behaved", since the Web IDL translation layer has already taken care of the kinks like getter functions, inheritance chain, etc. So here, you can simply do `init.is2D === false`.

Even better would be to use the same logic as https://drafts.fxtf.org/geometry/#matrix-validate-and-fixup, which deals with the following cases in order:

1. `init.is2D === true` but the matrix is not 2D: throw an error,
2. `init.is2D === undefined` and the matrix is not 2D: set `is2D` to false,
3. otherwise: set `is2D` to true.